<launch>

  <!-- Launch the line follower by default (set type argument) -->
  <arg name="type" default="line"/>

  <!-- Launch the parking controller on the real robot. Launches safety controller, controller node, cone detection node, and the homography transformer -->
  <node pkg="safety_controller" name="safety_controller" exec="safety_controller" output="screen">
    <param from="$(find-pkg-share visual_servoing)/config/$(var type)_params.yaml"/> 
  </node>

  <node pkg="visual_servoing" name="parking_controller" exec="parking_controller" output="screen">
    <param from="$(find-pkg-share visual_servoing)/config/$(var type)_params.yaml"/> 
  </node>

  <node pkg="visual_servoing" name="cone_detector" exec="cone_detector" output="screen">
    <param from="$(find-pkg-share visual_servoing)/config/$(var type)_params.yaml"/> 
  </node>

  <node pkg="visual_servoing" name="homography_transformer" exec="homography_transformer" output="screen">
    <param from="$(find-pkg-share visual_servoing)/config/$(var type)_params.yaml"/> 
  </node>

  <node pkg="image_transport" name="republish" exec="republish" args="raw in:=/cone_debug_img out:=/cone_debug_img_compressed" output="screen"></node>

</launch>
